analyses/MD_cd_2010/01_prep_MD_cd_2010.R

```diff
@@ -34,54 +34,76 @@ shp_path <- "data-out/MD_2010/shp_vtd.rds"
-    md_shp <- read_csv(here(path_data), col_types = cols(GEOID10 = "c")) %>%
-        join_vtd_shapefile(year = 2010) %>%
-        st_transform(EPSG$MD)  %>%
-        rename_with(function(x) gsub("[0-9.]", "", x), starts_with("GEOID"))
-
-    # add municipalities
-    d_muni <- make_from_baf("MD", "INCPLACE_CDP", "VTD", year = 2010)  %>%
-        mutate(GEOID = paste0(censable::match_fips("MD"), vtd)) %>%
-        select(-vtd)
-    d_cd <- make_from_baf("MD", "CD", "VTD", year = 2010)  %>%
-        transmute(GEOID = paste0(censable::match_fips("MD"), vtd),
-            cd_2000 = as.integer(cd))
-    md_shp <- left_join(md_shp, d_muni, by = "GEOID") %>%
-        left_join(d_cd, by = "GEOID") %>%
-        mutate(county_muni = if_else(is.na(muni), county, str_c(county, muni))) %>%
-        relocate(muni, county_muni, cd_2000, .after = county)
-
-    # add the enacted plan
-    cd_shp <- st_read(here(path_enacted))
-    md_shp <- md_shp %>%
-        mutate(cd_2010 = as.integer(cd_shp$DISTRICT)[
-            geo_match(md_shp, cd_shp, method = "area")],
-        .after = cd_2000)
-
-    # Create perimeters in case shapes are simplified
-    redistmetrics::prep_perims(shp = md_shp,
-        perim_path = here(perim_path)) %>%
-        invisible()
-
-    # simplifies geometry for faster processing, plotting, and smaller shapefiles
-    if (requireNamespace("rmapshaper", quietly = TRUE)) {
-        md_shp <- rmapshaper::ms_simplify(md_shp, keep = 0.05,
-            keep_shapes = TRUE) %>%
-            suppressWarnings()
-    }
-
-    # create adjacency graph
-    md_shp$adj <- redist.adjacency(md_shp)
-
-    md_shp <- md_shp %>%
-        fix_geo_assignment(muni)
-
-    write_rds(md_shp, here(shp_path), compress = "gz")
-    cli_process_done()
+  cli_process_start("Preparing {.strong MD} shapefile")
+  # read in redistricting data
+  md_shp <- read_csv(here(path_data), col_types = cols(GEOID10 = "c")) %>%
+    join_vtd_shapefile(year = 2010) %>%
+    st_transform(EPSG$MD)  %>%
+    rename_with(function(x) gsub("[0-9.]", "", x), starts_with("GEOID"))
+  
+  # add municipalities
+  d_muni <- make_from_baf("MD", "INCPLACE_CDP", "VTD", year = 2010)  %>%
+    mutate(GEOID = paste0(censable::match_fips("MD"), vtd)) %>%
+    select(-vtd)
+  d_cd <- make_from_baf("MD", "CD", "VTD", year = 2010)  %>%
+    transmute(GEOID = paste0(censable::match_fips("MD"), vtd),
+              cd_2000 = as.integer(cd))
+  md_shp <- left_join(md_shp, d_muni, by = "GEOID") %>%
+    left_join(d_cd, by = "GEOID") %>%
+    mutate(county_muni = if_else(is.na(muni), county, str_c(county, muni))) %>%
+    relocate(muni, county_muni, cd_2000, .after = county)
+  
+  # add the enacted plan
+  cd_shp <- st_read(here(path_enacted))
+  md_shp <- md_shp %>%
+    mutate(cd_2010 = as.integer(cd_shp$DISTRICT)[
+      geo_match(md_shp, cd_shp, method = "area")],
+      .after = cd_2000)
+  
+  # Exclude bay areas.
+  bay_ids <- c("24037ZZZZZZ", "24009ZZZZZZ", "24041ZZZZZZ")
```

Comment:
As with the 2000 data, can we perform this by removing edges from the graph, rather than editing the underlying shape? This matters for downstream users who may want to add/join other data to these shapes.